vxyowsup/tests/test_whatsapp.py

```diff
@@ -26,8 +26,12 @@ def getDummyCoreLayers():
 def TUMessage_to_PTNode(message):
     # message is TransportUserMessage
     # returns ProtocolTreeNode
-    return TextMessageProtocolEntity(message['content'], to=message['to_addr']
-                                     + '@s.whatsapp.net').toProtocolTreeNode()
+    if message['to_addr']:
+        return TextMessageProtocolEntity(message['content'], to=message['to_addr']
+                                         + '@s.whatsapp.net').toProtocolTreeNode()
+    elif message['from_addr']:
```

Comment:
I think it should usually be the case that both `to_addr` and `from_addr` are set, so perhaps we should structure this as:
```python
to_addr = add_whatsapp_suffix(message['to_addr'])
from_addr = add_whatsapp_suffix(message['from_addr'])
return TextMessageProtocolEntity(message['content'], to=to_addr, _from=from_addr)
```
where
```python
def add_whatsapp_suffix(addr):
    if addr:
        addr = addr + '@s.whatsapp.net'
    return addr
```
Thoughts?